### PR TITLE
Tests for TensorSliceDataset and SparseTensorSliceDataset

### PR DESCRIPTION
**Pull request by feihugis**
This PR adds the tests for TensorSliceDataset and SparseTensorSliceDataset. As #25496 is not merged yet, there are two duplicated util functions (`CreateTensor()` and `ExpectEqual()`) in this PR. Once either of them is merged, will update the other one. 

cc @jsimsa

### CODE REVIEW

File: tensorflow/core/kernels/data/sparse_tensor_slice_dataset_op_test.cc [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+
+class SparseTensorSliceDatasetOpTest : public DatasetOpsTestBase {
+ protected:
+  // Creates a new SparseTensorSliceDataset op kernel.
+  Status CreateSparseTensorSliceDatasetKernel(
+      DataType tvalues, std::unique_ptr<OpKernel> *op_kernel) {
+    node_def_ = test::function::NDef(kNodeName, kOpName,
+                                     {"indices", "values", "dense_shape"},
+                                     {{"Tvalues", tvalues}});
+    TF_CHECK_OK(CreateOpKernel(node_def_, op_kernel));
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/sparse_tensor_slice_dataset_op_test.cc]: TF_RETURN_IF_OK?

File: tensorflow/core/kernels/data/sparse_tensor_slice_dataset_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+    TF_ASSERT_OK(iterator->Restore(iterator_ctx.get(), &reader));
+  }
+}
+
+INSTANTIATE_TEST_CASE_P(SparseTensorSliceDatasetOpTest, IteratorRoundtripTest,
+                        ::testing::ValuesIn(TestCases));
+
+}  // namespace
+}  // namespace data
+}  // namespace tensorflow
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/sparse_tensor_slice_dataset_op_test.cc]: missing new line

File: tensorflow/core/kernels/data/tensor_slice_dataset_op_test.cc [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+    std::vector<string> components;
+    components.reserve(dtypes.size());
+    for (int i = 0; i < dtypes.size(); i++) {
+      components.emplace_back(strings::StrCat("component_", i));
+    }
+
+    node_def_ = test::function::NDef(
+        kNodeName, kOpName, components,
+        {{"Toutput_types", dtypes}, {"output_shapes", shapes}});
+    TF_CHECK_OK(CreateOpKernel(node_def_, tensor_dataset_kernel));
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/tensor_slice_dataset_op_test.cc]: TF_RETURN_IF_ERROR?

File: tensorflow/core/kernels/data/tensor_slice_dataset_op_test.cc [hunk 4]
```diff
@@ -0,0 +1,630 @@
...
+    TF_ASSERT_OK(iterator->Restore(iterator_context.get(), &reader));
+  }
+}
+
+INSTANTIATE_TEST_CASE_P(TensorDatasetSliceOpTest, IteratorRoundtripTest,
+                        ::testing::ValuesIn(TestCases));
+
+}  // namespace
+}  // namespace data
+}  // namespace tensorflow
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/tensor_slice_dataset_op_test.cc]: missing new line

**Review by jsimsa — CHANGES_REQUESTED** [review]
Thanks! Left some minor comments. Overall this looks great.

**Comment by feihugis** [comment]
@jsimsa Thanks for your quick review! [This commit](https://github.com/tensorflow/tensorflow/pull/25599/commits/1f711cc6f11fdf0d528f4d20c70fd7b9713c72bf) replaces `TF_CHECK_OK` with `TF_RETURN_IF_ERROR` and adds the missing new line. Could you have a look at the changes?

**Review by jsimsa — APPROVED** [review]

**Comment by feihugis** [comment]
The test failure is caused by `//bazel_pip/tensorflow/contrib/gan:classifier_metrics_test`, so it is unrelated.

**Pull request merged**